MekHQ/src/mekhq/campaign/unit/Unit.java

```diff
@@ -3871,4 +3871,244 @@ public String toString() {
+        long partsCost = 0;
+
+        entity = getEntity();
+        if (isMothballed()) {
+            return partsCost;
+        }
+        if ((entity instanceof Jumpship) || (entity instanceof SpaceStation)) {
+            partsCost += ((long)entity.getWeight()) * .0001 * 15000;
+        } else if (entity instanceof Aero) {
+            partsCost += ((long)entity.getWeight()) * .001 * 15000;
+        } else if (entity instanceof Tank) {
+            partsCost += ((long)entity.getWeight()) * .001 * 8000;
+        } else if ((entity instanceof Mech) || (entity instanceof BattleArmor)
+                || ((Infantry) entity).isMechanized()) {
+            partsCost += ((long)entity.getWeight()) * .001 * 10000;
+        } else if (entity instanceof Infantry) {
+            if (entity.getMovementMode() == EntityMovementMode.INF_LEG) {
+                partsCost += 3 * .002 * 10000;
+            } else if (entity.getMovementMode() == EntityMovementMode.INF_JUMP) {
+                partsCost += 4 * .002 * 10000;
+            } else if (entity.getMovementMode() == EntityMovementMode.INF_MOTORIZED) {
+                partsCost += 6 * .002 * 10000;
+            } else {
+                partsCost += ((long)entity.getWeight()) * .002 * 10000;
+                MekHQ.getLogger().log(getClass(), METHOD_NAME, LogLevel.ERROR,
+                        getName() + " is not a generic CI. Movement mode is "
+                        + entity.getMovementModeAsString());
+            }
+
+        } else {
+            // Only protomechs should fall here. Anything else needs to be logged
+            if (!(entity instanceof Protomech)) {
+                MekHQ.getLogger().log(getClass(), METHOD_NAME, LogLevel.ERROR,
+                        getName() + " has no Spare Parts value for unit type " 
+                        + Entity.getEntityTypeName(entity.getEntityType()));
+            }
+        }
+
+        // Handle cost for quirks if used
+        if (entity.hasQuirk("easy_maintain")) {
+            partsCost = (long)(partsCost * .8);
+        }
+        if (entity.hasQuirk("difficult_maintain")) {
+            partsCost = (long)(partsCost * 1.25);
+        }
+        if (entity.hasQuirk("non_standard")) {
+            partsCost = (long)(partsCost * 2.0);
+        }
+        if (entity.hasQuirk("ubiquitous_is")) {
+            partsCost = (long)(partsCost * .75);
+        }
+        // TODO Obsolete quirk
+
+        // Now for extended parts cost modifiers
+        if (campaign.getCampaignOptions().useExtendedPartsModifier()) {
+            Engine en = entity.getEngine();
+            int currentYear = campaign.getCalendar().get(Calendar.YEAR);
+            int rating = getTechRating();
+            if ((currentYear > 2859) && (currentYear < 3040)) {
+                if (rating > 3) {
+                    partsCost = (long)(partsCost * 5.0);
+                }
+            }
+            if (rating == 4) {
```

Comment:
EquipmentType.RATING_E
This will be cumulative with the x5 multiplier. While you could read the chart that way, I think the intention is that E is x1.1 and F is x1.25 unless it's a non-Clan, non-ComStar unit, in which case it's x5 instead.